dotnet/orleans
# Clean up Orleans.2017.sln

Second attempt of: Clean up Orleans.2017.sln (#3038)

First iteration towards the merging of the source trees. This one cleans up the Orleans.2017.sln which we use to build the main tree:

- Update F# to 4.1.17 - the compiler and core versions are aligned by the F# team
- Update xUnit to 2.2.0
- Add F# projects to Orleans.2017 Solution (only dotnet.exe build can build it, excluded when building in VS, hence 2 projects are not loading)
- Update VSWhere to latest version
- Remove single output folder requirement for build
- Update 2017 project files to share common properties, enable assemblyinfo generation and nuget packing, removed unused things
- Create 2017 version of Build and Test batch files
- Enable /m for Build.2017.cmd
- Update Skippablefacts package to work with 2.2 xUnit

New in this PR:
Add ImportDirectoryProps and target = false to src/build.cmd to make sure new files are not picked up by old builds.
Add empty Directory.Build.props for VSIX for the same reason.
